features/data_converter/empty/feature.ts

```diff
@@ -0,0 +1,44 @@
+import { JSONToPayload } from '@temporalio/common/lib/proto-utils';
+import { Feature } from '@temporalio/harness';
+import * as assert from 'assert';
+import { ApplicationFailure } from '@temporalio/common';
+import expectedPayload from './payload.json';
+import * as wf from '@temporalio/workflow';
+
+const activitiesImpl = {
+  async activity(input: any): Promise<void> {
+    if (input != undefined) {
+      throw ApplicationFailure.nonRetryable('Activity input should be undefined', 'BadResult');
```

Comment:
nit: the type is optional and `BadResult` is not really a good name for the error.
Not blocking this PR.